R/get_wb_data.R

```diff
@@ -0,0 +1,51 @@
+#' Downloads all available data from one or more datasets from the World Bank database.
+#'
+#' \code{get_wb_data} generalizes the function \code{WDI::WDI()} to download the whole dataset instead of single indicators. It defaults to a starting date of 1960 up to 2022. It returns a list of data.frames.
+#'
+#' @param datasets a \code{wb_data} object or a vector of strings containing the names of the datasets to be downloaded.
+#' @param start_year starting year to pull data from. Must be 1960 or higher.
+#' @param end_year last year to pull data from. Must be higher than start_year.
+#'
+#' @section Warning:
+#' An internet connection is needed to use this function.
+#' 
+#' @examples \dontrun{
+#' library("r-epigraphhub")
+#' datasets <- wb_data()
+#' my_dataset <- datasets[2]
+#' my_data <- get_wb_data(my_dataset)
+#'
+#' my_dataset <- "Environment, Social and Governance (ESG) Data"
+#' my_data <- get_wb_data(my_dataset)
+#' }
+#' @export
+
+get_wb_data <- function(datasets, start_year = 1960, end_year = 2022){
+  require(dplyr)
```

Comment:
Wouldn't it be better to move these `requires` outside the function? As it is now every time the function is called the requires will be run.

Reply:
Done in the newer commit. Instead, all outside functions are listed in the NAMESPACE file.